worker/migrationmaster/worker.go

```diff
@@ -281,8 +285,13 @@ func (w *Worker) doQUIESCE(status coremigration.MigrationStatus) (coremigration.
 }
 
 func (w *Worker) doPRECHECK() (coremigration.Phase, error) {
-	// TODO(mjs) - To be implemented.
-	// w.setInfoStatus("performing prechecks")
+	w.setInfoStatus("performing prechecks")
+	err := w.config.Facade.Prechecks()
+	if err != nil {
+		w.setErrorStatus("prechecks failed, %v", err)
+		return coremigration.ABORT, nil
+	}
+	// TODO(mjs) - perform prechecks on target controller
```

Comment:
Yep and a bunch of other things such as the health of the controller machines, that the controller isn't upgrading.